### PR TITLE
Add func to check the expect words

### PR DESCRIPTION
**Pull request by juteman**
the func can check the expect match or
expect not match

Signed-off-by: qiulzhan <qiulzhan@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
Looks good to me

File: virttest/utils_test/libvirt.py [hunk 1]
```diff
@@ -3577,3 +3578,14 @@ def get_disk_alias(vm, source_file=None):
                 logging.info("Ignore error of source attr getting for file: %s" % e)
                 pass
     return None
+
+
+def check_cmd_expected(check_cmd, expected_match, is_expected):
```
Comment (kylazhang) [inline virttest/utils_test/libvirt.py]: Function explanation and parameters explanation.

Comment (kylazhang) [reply virttest/utils_test/libvirt.py]: And you can consider more scenario, like not expected match

Comment (juteman) [reply virttest/utils_test/libvirt.py]: thks for advice. Add commend and add a new scenario

File: virttest/utils_test/libvirt.py [hunk 2]
```diff
@@ -3577,3 +3578,26 @@ def get_disk_alias(vm, source_file=None):
...
+                                               shell=True))
+
+    if matched and not re.search(matched, out):
+        if is_expected:
+            raise exceptions.TestError(" '%s' not found in output: %s " %
+                                       (matched, out))
+
+    elif matched and re.search(matched, out):
+        if not is_expected:
+            raise exceptions.TestError(" '%s' found but expected no found in output: %s" %
```
Comment (kylazhang) [inline virttest/utils_test/libvirt.py]: Better description for raise error: "%s is not expected but found in output: %s"

File: virttest/utils_test/libvirt.py [hunk 3]
```diff
@@ -3577,3 +3578,26 @@ def get_disk_alias(vm, source_file=None):
...
+
+
+def check_cmd_expected(check_cmd, matched, is_expected):
+    """
+    check the cmd result that process run
+
+    Params check_cmd: the command need to run
+    Params matched: need to match or not match result
+    Params is_expected:
+    """
```
Comment (kylazhang) [inline virttest/utils_test/libvirt.py]: Format need update and "is_expected" need to explain

File: virttest/utils_test/libvirt.py [hunk 4]
```diff
@@ -3577,3 +3578,26 @@ def get_disk_alias(vm, source_file=None):
...
                 pass
     return None
+
+
+def check_cmd_expected(check_cmd, matched, is_expected):
+    """
+    check the cmd result that process run
+
+    Params check_cmd: the command need to run
+    Params matched: need to match or not match result
```
Comment (kylazhang) [inline virttest/utils_test/libvirt.py]: "expected or not expected match pattern"

File: virttest/utils_test/libvirt.py [hunk 5]
```diff
@@ -3577,3 +3578,26 @@ def get_disk_alias(vm, source_file=None):
                 logging.info("Ignore error of source attr getting for file: %s" % e)
                 pass
     return None
+
+
+def check_cmd_expected(check_cmd, matched, is_expected):
```
Comment (kylazhang) [inline virttest/utils_test/libvirt.py]: Better give default value for is_expected

File: virttest/utils_test/libvirt.py [hunk 6]
```diff
@@ -3577,3 +3578,26 @@ def get_disk_alias(vm, source_file=None):
...
+
+    Params check_cmd: the command need to run
+    Params matched: need to match or not match result
+    Params is_expected:
+    """
+    out = decode_to_text(process.system_output(check_cmd,
+                                               ignore_status=False,
+                                               shell=True))
+
+    if matched and not re.search(matched, out):
```
Comment (kylazhang) [inline virttest/utils_test/libvirt.py]: here I think the "matched" must be given, so no need to do judgement here

File: virttest/utils_test/libvirt.py (hunk 3 above)
Comment (kylazhang) [reply virttest/utils_test/libvirt.py]: And better give the type for it, boolean or str

File: virttest/utils_test/libvirt.py (hunk 6 above)
Comment (kylazhang) [reply virttest/utils_test/libvirt.py]: better logic is:
```
if is_expected:
    if not re.search():
        raise xxx
    else:
         logging.info()
else:
    if re.search():
        raise xxx
    else:
        logging.info()
```

Comment (juteman) [reply virttest/utils_test/libvirt.py]: the match check here is that i think some time we may check the result as  "blank"

File: virttest/utils_test/libvirt.py (hunk 3 above)
Comment (juteman) [reply virttest/utils_test/libvirt.py]: add type to commond

File: virttest/utils_test/libvirt.py [hunk 7]
```diff
@@ -3577,3 +3578,40 @@ def get_disk_alias(vm, source_file=None):
                 logging.info("Ignore error of source attr getting for file: %s" % e)
                 pass
     return None
+
+
+def check_cmd_expected(check_cmd, matched, is_expected=True):
```
Comment (kylazhang) [inline virttest/utils_test/libvirt.py]: Better give matched default value to None

File: virttest/utils_test/libvirt.py [hunk 8]
```diff
@@ -3577,3 +3578,40 @@ def get_disk_alias(vm, source_file=None):
...
+                                               ignore_status=False,
+                                               shell=True))
+
+    # if we need to match "nothing"
+    if not matched:
+        if out:
+            raise exceptions.TestError("the command result is: \n %s \n"
+                                       "but expected no result" % out)
+        else:
+            logging.info("command has no result")
```
Comment (kylazhang) [inline virttest/utils_test/libvirt.py]: result->output

File: virttest/utils_test/libvirt.py [hunk 9]
```diff
@@ -3577,3 +3578,40 @@ def get_disk_alias(vm, source_file=None):
...
+    """
+    out = decode_to_text(process.system_output(check_cmd,
+                                               ignore_status=False,
+                                               shell=True))
+
+    # if we need to match "nothing"
+    if not matched:
+        if out:
+            raise exceptions.TestError("the command result is: \n %s \n"
+                                       "but expected no result" % out)
```
Comment (kylazhang) [inline virttest/utils_test/libvirt.py]: result->output for above 2 place

File: virttest/utils_test/libvirt.py (hunk 7 above)
Comment (balamuruhans) [inline virttest/utils_test/libvirt.py]: we can reuse the existing API,
https://github.com/avocado-framework/avocado-vt/blob/e07894bf447cf83ff34eeffecc68ebb99a5d7892/virttest/utils_test/libvirt.py#L1514-L1532

Comment (juteman) [reply virttest/utils_test/libvirt.py]: yes，i think so.

Comment (juteman) [reply virttest/utils_test/libvirt.py]: I will close this pr

**Pull request closed without merging**